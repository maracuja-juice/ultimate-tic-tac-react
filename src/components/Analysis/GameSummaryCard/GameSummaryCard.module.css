.padding10 {
  padding: 10px;
}

.paddingBot10 {
  padding-bottom: 10px;
}

.italic {
  font-style: italic;
}
