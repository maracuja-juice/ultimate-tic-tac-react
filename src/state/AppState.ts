import { Point } from '../util';

export interface AppState {
  currentGame: GameInformation;
  finishedGames: FinishedGameState[];
  analysisGame: AnalysisGame;
}

export interface GameInformation {
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
}

export interface FinishedGameState {
  id?: string;
  date: string;
  winner: Winner;
  gameState: SmallBoardInformation[];
  moves: MoveState[];
  saveState: string;
  errorMessage: string;
}

export interface AnalysisGame {
  id: string;
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
  currentMove: number;
}

export interface GameState {
  currentPlayer: Player;
  isFinished: boolean;
  winningPlayer: Winner;
}

export interface TileInformation {
  position: Point;
  value: TileValue;
}

export interface SmallBoardInformation extends TileInformation {
  tiles: SmallTileInformation[];
}

export interface SmallTileInformation extends TileInformation {
  boardPosition: Point;
}

export enum TileValue {
  Cross = 0,
  Circle = 1,
  Empty = 2,
  Destroyed = 3,
}

export enum Winner {
  Cross = 0,
  Circle = 1,
  None = 2,
  Draw = 3,
}

export enum Player {
  Cross = 0,
  Circle = 1,
}

export interface Move {
  boardPosition: Point;
  tilePosition: Point;
}

export interface MoveState extends Move {
  player: Player;
  moveNumber: number;
}

export interface GenericAction {
  type: string;
  // tslint:disable-next-line: no-any
  payload?: any;
}

export interface MarkSpecially {
  condition: boolean;
  position?: { boardPosition: Point; tilePosition: Point };
}
