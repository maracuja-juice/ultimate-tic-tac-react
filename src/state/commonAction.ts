export const RESTART_GAME = 'common/restart-game';

export const restartGame = () => ({
    type: RESTART_GAME
});