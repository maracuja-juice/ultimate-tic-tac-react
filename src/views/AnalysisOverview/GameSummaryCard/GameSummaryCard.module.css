.winnerIcon {
  margin-right: 3px;
}

.padding10 {
  padding: 10px;
}

.paddingBot10 {
  padding-bottom: 10px;
}

.italic {
  font-style: italic;
}
