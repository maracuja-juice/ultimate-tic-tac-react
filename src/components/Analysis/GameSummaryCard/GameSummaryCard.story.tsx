import { storiesOf } from '@storybook/react';
import { date, number, select, text, withKnobs } from '@storybook/addon-knobs';
import * as React from 'react';
import { GameSummaryCard } from './GameSummaryCard';
import {
  circleFinishedBoardMock,
  crossFinishedBoardMock,
  drawFinishedBoardMock,
  movesForCircleFinishedBoardMock,
  movesForCrossFinishedBoardMock,
  movesForDrawFinishedBoardMock,
} from '../../../__mocks__';
import { Link, BrowserRouter as Router } from 'react-router-dom';
import { WinnerString } from '../../../state/AppState';
import { IconProvider } from '../../IconProvider';

const stories = storiesOf('Analysis', module);
stories.addDecorator(withKnobs);

const gameStateOptions = {
  cross: {
    state: crossFinishedBoardMock,
    moves: movesForCrossFinishedBoardMock,
    winner: 'X',
  },
  circle: {
    state: circleFinishedBoardMock,
    moves: movesForCircleFinishedBoardMock,
    winner: 'O',
  },
  draw: {
    state: drawFinishedBoardMock,
    moves: movesForDrawFinishedBoardMock,
    winner: null,
  },
};

function isoStringDateKnob(name: string, defaultValue: Date) {
  const stringTimestamp = date(name, defaultValue);
  return new Date(stringTimestamp).toISOString();
}

stories.add('GameSummaryCard', () => {
  const selectedGameState = select(
    'gameState',
    gameStateOptions,
    gameStateOptions.cross,
  );

  const defaultDate = new Date(2019, 1, 1);

  return (
    <IconProvider>
      <Router>
        <div style={{ height: '100%' }}>
          <GameSummaryCard
            gameNumber={number('gameNumber', 1)}
            game={{
              id: '1',
              gameState: selectedGameState.state,
              date: isoStringDateKnob('date', defaultDate),
              winner: selectedGameState.winner as WinnerString,
              moves: selectedGameState.moves,
              saveState: 'fulfilled',
              errorMessage: '',
            }}
            link={{ tag: Link, to: text('to', '/test') }}
          />
        </div>
      </Router>
    </IconProvider>
  );
});
