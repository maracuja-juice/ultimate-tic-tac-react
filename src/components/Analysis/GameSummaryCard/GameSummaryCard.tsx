import * as React from 'react';
import { BigBoard } from '../../Board/BigBoard/BigBoard';
import {
  Card,
  CardPrimaryAction,
  CardActions,
  CardActionButton,
} from '@rmwc/card';
import { formatDistanceStrict } from 'date-fns';
import { FinishedGameState } from '../../../state/AppState';
import { Typography } from '@rmwc/typography';
import { ListDivider } from '@rmwc/list';
import { Icon } from '@rmwc/icon';
import { Link } from 'react-router-dom';
import styles from './GameSummaryCard.module.css';

interface GameSummaryCardProps {
  gameNumber: number;
  game: FinishedGameState;
  link: { tag: typeof Link; to: string };
}

function getGameSummary(winner: 'X' | 'O' | null, moves: number) {
  if (winner) {
    return (
      <>
        <Icon
          icon={{
            icon: winner.toLowerCase(),
            size: 'medium',
          }}
          className={styles.winnerIcon}
        />
        won after {moves} moves.
      </>
    );
  }

  return <>{moves} moves resulted in a draw.</>;
}

export function GameSummaryCard(props: GameSummaryCardProps) {
  const { game, gameNumber, link } = props;

  // TODO add storybook
  // TODO add tests?!!

  return (
    <Card>
      <CardPrimaryAction className={styles.padding10} {...link}>
        <div className={styles.padding10}>
          <div>
            <Typography use={'headline4'} tag="h2">
              Game No. {gameNumber}
            </Typography>
          </div>
          <div>
            <Typography use={'subtitle2'} tag="h3">
              {getGameSummary(game.winner, game.moves.length)}
            </Typography>
          </div>
          <div className={styles.paddingBot10}>
            <Typography use={'subtitle2'} tag="p" className={styles.italic}>
              {formatDistanceStrict(
                new Date(),
                new Date(Date.parse(game.date)),
              )}{' '}
              ago
            </Typography>
          </div>
          <ListDivider />
        </div>
        <BigBoard
          currentPlayer={game.moves[game.moves.length - 2].player}
          board={game.gameState}
          activeBoards={[]}
          onPlayerMoved={() => {}}
          animate={false}
        />
      </CardPrimaryAction>
      <CardActions>
        <CardActionButton
          label="Analyse Game"
          trailingIcon="arrow-right"
          {...link}
        />
      </CardActions>
    </Card>
  );
}
