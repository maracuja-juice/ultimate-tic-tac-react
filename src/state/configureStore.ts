import { AppState } from './AppState';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import tileReducer from './tiles/tileReducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import gameReducer from './game/gameReducer';
import moveReducer from './moves/moveReducer';
import { enableBatching } from 'redux-batched-actions';
import activeBoardReducer from './activeBoards/activeBoardsReducer';
import createSagaMiddleware from 'redux-saga';
import { all, fork } from 'redux-saga/effects';
import moveValidationSaga from './sagas/MoveValidationSaga';

const rootreducer = combineReducers<AppState>(
    {
        game: gameReducer,
        tiles: tileReducer,
        moves: moveReducer,
        activeBoards: activeBoardReducer
    } );

export function configureStore() {

    const sagaMiddleware = createSagaMiddleware();

    const middleWaresToApply = [
        logger,
        sagaMiddleware
    ];
    const middleware = applyMiddleware( ...middleWaresToApply );
    const store = createStore( enableBatching( rootreducer ), composeWithDevTools(
        middleware
    ) );

    sagaMiddleware.run( rootSaga );

    return store;
}

function* rootSaga() {
    yield all(
        [
            fork( moveValidationSaga ),
        ]
    );
}