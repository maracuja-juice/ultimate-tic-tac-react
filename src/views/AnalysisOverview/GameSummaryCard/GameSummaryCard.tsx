import * as React from 'react';
import { Card, CardActions, CardActionButton } from '@rmwc/card';
import { formatDistanceStrict } from 'date-fns';
import { FinishedGameState } from '../../../state/AppState';
import { Typography } from '@rmwc/typography';
import { ListDivider } from '@rmwc/list';
import { Icon } from '@rmwc/icon';
import { Link } from 'react-router-dom';
import styles from './GameSummaryCard.module.css';
import { ArrowRightIcon, OIcon, XIcon } from '../../../components/Icons';
import { BigBoard } from '../../../components/Board/BigBoard/BigBoard';

export interface GameSummaryCardProps {
  gameNumber: number;
  game: FinishedGameState;
  link: { tag: typeof Link; to: string };
}

function getGameSummary(winner: 'X' | 'O' | null, moves: number) {
  if (winner) {
    return (
      <>
        <Icon
          icon={{
            icon: winner === 'X' ? <XIcon /> : <OIcon />,
            size: 'small',
            'aria-label': winner,
          }}
          className={styles.winnerIcon}
        />
        won after {moves} moves.
      </>
    );
  }

  return <>{moves} moves resulted in a draw.</>;
}

export function GameSummaryCard(props: GameSummaryCardProps) {
  const { game, gameNumber, link } = props;

  return (
    <Card>
      <div className={styles.padding10}>
        <div>
          <Typography use={'headline4'} tag="h2">
            Game No. {gameNumber}
          </Typography>
        </div>
        <div>
          <Typography use={'subtitle2'} tag="h3">
            {getGameSummary(game.winner, game.moves.length)}
          </Typography>
        </div>
        <div className={styles.paddingBot10}>
          <Typography use={'subtitle2'} tag="p" className={styles.italic}>
            {formatDistanceStrict(new Date(), new Date(Date.parse(game.date)))}{' '}
            ago
          </Typography>
        </div>
        <ListDivider tag="div" />
      </div>
      <div className={styles.padding10}>
        <a className="sr-only" href={`#analyse-button-${gameNumber}`}>
          Skip Board Declaration
        </a>
        <BigBoard
          currentPlayer={game.moves[game.moves.length - 2].player}
          board={game.gameState}
          activeBoards={[]}
          animate={false}
        />
      </div>
      <CardActions>
        <CardActionButton
          id={`analyse-button-${gameNumber}`}
          label="Analyse Game"
          trailingIcon={{ icon: <ArrowRightIcon />, 'aria-hidden': true }}
          {...link}
        />
      </CardActions>
    </Card>
  );
}
