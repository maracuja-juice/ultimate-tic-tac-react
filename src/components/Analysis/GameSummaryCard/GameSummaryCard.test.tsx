import * as React from 'react';
import { shallow } from 'enzyme';
import { GameSummaryCard } from './GameSummaryCard';
import {
  circleFinishedBoardMock,
  movesForCircleFinishedBoardMock,
} from '../../../__mocks__';
import { Link } from 'react-router-dom';
import { SaveState, WinnerString } from '../../../state/AppState';

describe('GameSummaryCard', () => {
  it('should match snapshot', () => {
    const game = {
      id: '1',
      gameState: circleFinishedBoardMock,
      date: new Date(2019, 1, 1).toISOString(),
      winner: 'O' as WinnerString,
      moves: movesForCircleFinishedBoardMock,
      saveState: 'fulfilled' as SaveState,
      errorMessage: '',
    };

    const card = shallow(
      <GameSummaryCard
        gameNumber={1}
        game={game}
        link={{ tag: Link, to: '/a/location' }}
      />,
    );

    expect(card).toMatchSnapshot();
  });
});
