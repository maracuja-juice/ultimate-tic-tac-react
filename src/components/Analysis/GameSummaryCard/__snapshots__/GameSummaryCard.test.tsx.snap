// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSummaryCard should match snapshot 1`] = `
<Function>
  <div
    className="padding10"
  >
    <div>
      <Function
        tag="h2"
        use="headline4"
      >
        Game No. 
        1
      </Function>
    </div>
    <div>
      <Function
        tag="h3"
        use="subtitle2"
      >
        <Icon
          className="winnerIcon"
          icon={
            Object {
              "aria-label": "O",
              "icon": <MdPanoramaFishEye />,
              "size": "medium",
            }
          }
        />
        won after 
        30
         moves.
      </Function>
    </div>
    <div
      className="paddingBot10"
    >
      <Function
        className="italic"
        tag="p"
        use="subtitle2"
      >
        3 days
         
        ago
      </Function>
    </div>
    <Function
      tag="div"
    />
  </div>
  <div
    className="padding10"
  >
    <a
      className="sr-only"
      href="#analyse-button-1"
    >
      Skip Board Declaration
    </a>
    <BigBoard
      activeBoards={Array []}
      animate={false}
      board={
        Array [
          Object {
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 1,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 1,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 0,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 1,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 0,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
        ]
      }
      currentPlayer={0}
    />
  </div>
  <Function>
    <Function
      id="analyse-button-1"
      label="Analyse Game"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/a/location"
      trailingIcon={
        Object {
          "aria-hidden": true,
          "icon": <MdChevronRight />,
        }
      }
    />
  </Function>
</Function>
`;
